sampleChangerApp/src/converter.cpp

```diff
@@ -231,52 +244,101 @@ std::string converter::get_available_in_slot(std::string slot)
     return res;
 }
 
-std::string converter::get_slot_for_position(std::string position) {
-    return m_slot_for_each_position.at(position);
+std::string converter::get_slot_for_position(std::string position)
+{
+    std::vector<std::pair<std::string, std::string>>::iterator it =
+        std::find_if(v_slot_for_each_position.begin(), v_slot_for_each_position.end(),
+            [&](std::pair<std::string, std::string> pair) {return pair.first == position; });
+    if (it == v_slot_for_each_position.end())
+    {
+        throw std::out_of_range("\nSlot not found at given position");
+    }
```

Comment:
You use this pattern in a lot of places, is it possible to put it into a function?

Reply:
Yes, I put this and other things of this kind into functions.